platform/event/core_event_data.proto

```diff
@@ -0,0 +1,74 @@
+/**
+ * # Core Event Data
+ * Contains information about an event and its parents.
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.event;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2018 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "event_descriptor.proto";
+import "timestamp.proto";
+import "basic_types.proto";
+
+option java_package = "com.hedera.hapi.platform.event";
+// <<<pbj.java_package = "com.hedera.hapi.platform.event">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * Contains information about an event and its parents.
+ */
+message CoreEventData {
+  /**
+   * The software version of the node at the time the event was created.<br/>
+   * This software version SHALL exactly match the software version passed into the platform at construction.
+   */
+  proto.SemanticVersion version = 1;
+
+  /**
+   * The ID of the node that created the event.
+   */
+  int64 creator_id = 2;
```

Comment:
Looking through the other protobufs, it seems node ID is always represented by an `int64`

I'm not sure whether this should be addressed now, since these *are* completely new definitions. @jasperpotts Do you have any thoughts on this?

Reply:
I think the platform's concept of a node ID is different from Hedera's. Sharding does not seem like something that we will be rolling out soon, so it would be hard to define if that should be part of the ID